x-pack/elastic-agent/pkg/artifact/download/http/downloader.go

```diff
@@ -132,6 +132,12 @@ func (e *Downloader) downloadFile(ctx context.Context, artifactName, filename, f
 		return "", errors.New(err, "fetching package failed", errors.TypeNetwork, errors.M(errors.MetaKeyURI, sourceURI))
 	}
 
+	destinationFile, err := os.OpenFile(fullPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, packagePermissions)
```

Comment:
This was moved to take the local file into account first? Not sure I fully follow this part here.

Reply:
I think it was to ensure that the file path to write to can be opened before even starting the HTTP connection.

Reply:
yes exactly, just a micro optimization